streams/src/test/java/org/apache/kafka/streams/kstream/KStreamBuilderTest.java

```diff
@@ -22,12 +22,23 @@
 import org.apache.kafka.streams.errors.TopologyBuilderException;
 import org.apache.kafka.test.KStreamTestDriver;
 import org.apache.kafka.test.MockProcessorSupplier;
+import org.junit.After;
 import org.junit.Test;
 
 import static org.junit.Assert.assertEquals;
 
 public class KStreamBuilderTest {
 
+    private KStreamTestDriver driver = null;
+
+    @After
+    public void cleanup() {
+        if (driver != null) {
+            driver.close();
+        }
+        driver = null;
+    }
```

Comment:
@guozhangwang: Not sure I understand.  Can't we use a `@Rule` (= per-test `ExternalResource` setup; unlike a `@ClassRule`) for `KStreamTestDriver`, which I understand provides each test case its own resource, which can then be modified as needed (e.g. callling `driver.setTime()`)?